google/stackdriver_monitoring/install_monitoring.sh

```diff
@@ -0,0 +1,204 @@
+
+	   --period=60
+	        Number of seconds between pollings of microservices.
+
+	   --service_hosts <ip>+
+	        A comma-delimited list of hostnames (or IPs) to poll by default.
+	        The default is localhost.
+	        Make this empty "" to not poll any services by default.
+
+
+	<service_options> are in the form --<service>=<netloc>* where:
+	    <service> is one of clouddriver, echo, fiat, front50, gate, igor, rosco
+	    <netloc> is a comma-delimited list of either a <host> or <host>:<port>
+	    If only a <host> is provided, then the dfeault port will be used.
+
+	    An empty <netloc> list will disable polling on the service entirely.
+	    A value of "*" refers to all the --service_hosts.
+	    The default <netloc> for each of the services is "*".
+	   
+
+	<provider_options> are zero or more of:
+	    --credentials_path=<path>
+	        If using --stackdriver, the path for the Google Credentials to use.
+	        The default will be the application default credentials.
+EOF
+}
+
+
+function process_args() {
+  while [[ $# > 0 ]]
+  do
+      local key="$1"
+      shift
+      case $key in
+          --datadog)
+              USE_DATADOG=true
+              PROVIDERS="$PROVIDERS --datadog"
+              ;;
+
+          --prometheus)
+              USE_PROMETHEUS=true
+              PROVIDERS="$PROVIDERS --prometheus"
+              ;;
+
+          --stackdriver)
+              USE_STACKDRIVER=true
+              PROVIDERS="$PROVIDERS --stackdriver"
+              ;;
+
+          --help|-h)
+              print_usage
+              exit 1
+              ;;
+
+          *)
+              ;;  # ignore
+
+      esac
+  done
+}
+
+
+function install_dependencies() {
+  apt-get update
+  apt-get install python-pip python-dev -y
+  pip install -r $SOURCE_DIR/requirements.txt
+}
+
+
+function install_metric_services() {
+  if [[ "$USE_DATADOG" == "true" ]]; then
+      $SOURCE_DIR/config/datadog/install.sh
+  fi
+  if [[ "$USE_PROMETHEUS" == "true" ]]; then
+      $SOURCE_DIR/config/prometheus/install.sh
+  fi
+  if [[ "$USE_STACKDRIVER" == "true" ]]; then
+      local credentials=""
+      for arg in ${COMMAND_LINE_FLAGS[@]}; do
+          if [[ $arg = --credentials_path=* ]]; then
+              credentials=$arg
+          fi
+      done
+      $SOURCE_DIR/config/stackdriver/install.sh $credentials
+  fi
+}
+
+
+function write_startup_script() {
+  cat <<-EOF > "$SOURCE_DIR/monitor_spinnaker.sh"
+	#!/bin/bash
+
+	set -o allexport
+	if [[ -f /etc/default/spinnaker ]]; then
+	  source /etc/default/spinnaker
+	fi
+	if [[ -f "$SOURCE_DIR/environ" ]]; then
+	  source "$SOURCE_DIR/environ"
+	fi
+	set +o allexport
+
+	PYTHONWARNINGS=once \
+	python "$SOURCE_DIR/monitor_spinnaker.zip" \
+	monitor $@ "\$@"
+EOF
+  chmod 755 "$SOURCE_DIR/monitor_spinnaker.sh"
+}
+
+
+function write_upstart_script() {
+  cat <<-EOF > /etc/init/monitor_spinnaker.conf
+        start on filesystem or runlevel [2345]
```

Comment:
Indentation of `start` doesn't match `exec`. I don't know if that matters or not for `.conf` files.